rust/src/smb/detect.rs

```diff
@@ -201,3 +204,258 @@ pub extern "C" fn rs_smb_tx_get_dce_iface(state: &mut SMBState,
     }
     return 0;
 }
+
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_smb_cmd_match(
+    tx: &mut SMBTransaction, cmd_data: &mut SmbCmdData,
+) -> u8 {
+
+    let version = tx.vercmd.get_version();
+    let cmd;
+    if version == 1 {
+        cmd = tx.vercmd.get_smb1_cmd().1 as u16;
```

Comment:
`as` is not the best rust (even if it is used in Suricata).
Maybe `into` is better ?

Reply:
Done in https://github.com/OISF/suricata/pull/6980/commits/4df4a658784158ce465bc2869a04d080af3d8331

Reply:
thanks